### ISSUE TITLE
Warning CS4014 not triggered on async methods behind an interface within an Action or Func<>

### ISSUE THREAD

**Issue by XHale84** [dotnet/roslyn#64964]
Hello there, I hope I am right in this section.
If not, please move this issue to the right place.
Thanks in advance!

**brief description**

When calling an async method from within an Action or Func<> via an interface, CS4014 isn't triggered correctly.

Describe your **code style** rule here.

Awaitable methods should always trigger a CS4014 when they are not awaited (or handled synchronously), no matter where they are called in code.

**Languages applicable:**

Definitely C#, VB maybe?

**Code example that the analyzer should report:**

Setup:
```
public interface IDoStuff
    {  
        Task DoStuff();  
        Task DoStuffAsync();  
    }
    public class DoingThings : IDoStuff
    {
        public Task DoStuff()
        {
            return Task.Delay(1000);
            //Do stuff
        }
        public async Task DoStuffAsync() => await DoStuff();
    }
```
My CS4014 results are this:
```
DoingThings Obj = new DoingThings();
IDoStuff IObj = new DoingThings();
Obj.DoStuff(); //<--warning
Obj.DoStuffAsync();//<--warning
IObj.DoStuff();//<--warning
IObj.DoStuffAsync();//<--warning
Action action1 = () => Obj.DoStuff();//<--NO warning
Action action2 = () => Obj.DoStuffAsync();//<--warning
Action Iaction1 = () => IObj.DoStuff();//<--NO warning
Action Iaction2 = () => IObj.DoStuffAsync();//<--NO warning  
```
But they should all be warned IMO

I hope you get the idea.
In the passt only async methods triggered CS4014.
But in my example also return type Task is warned.

Thanks
